src/api/wiki.js

```diff
@@ -0,0 +1,333 @@
+                    let result;
+                    if (resp && 200 == resp.statusCode) {
+                        result = body;
+                    } else {
+                        result = '{}';
+                    }
+                    next(result);
+                });
+            } catch (e) {
+                next('{}');
+            }
+        })
+    );
+}
+
+/**
+ * Retrieve the full uri of a given wiki page.
+ *
+ * @param {string} path
+ * @param {string} query
+ * @return {string}
+ */
+function uri(path, query) {
+    const parts = url.parse(this.env.url);
+    var out = parts.protocol + '//' + parts.host + util.preparePath(path);
+    if (query) {
+        out += '?' + query;
+    }
+    return out;
+}
+
+/**
+ * Inserts a pages sub tree
+ * if reverse is non-zero, the sort is backward
+ * if ordered is true, the output is an <ol> instead of <ul>
+ *
+ * Special note: If ordered is true, pages whose locale differ from
+ * the current page's locale are omitted, to work around misplaced
+ * localizations showing up in navigation.
+ *
+ * @param {string} path
+ * @param {number} depth
+ * @param {number|boolean} self
+ * @param {number|boolean} reverse
+ * @param {number|boolean} ordered
+ */
+async function tree(path, depth, self, reverse, ordered) {
+    // If the path ends with a slash, remove it.
+    if (path.substr(-1, 1) === '/') {
+        path = path.slice(0, -1);
+    }
+
+    var pages = await this.page.subpages(path, depth, self);
+
+    if (reverse == 0) {
+        pages.sort(alphanumForward);
+    } else {
+        pages.sort(alphanumBackward);
+    }
+
+    return process_array(null, pages, ordered != 0, this.env.locale);
+
+    /**
+     * @param {string} t
+     * @return {string[]}
+     */
+    function chunkify(t) {
+        var tz = [],
+            x = 0,
+            y = -1,
+            n = 0,
+            i,
+            j;
+
+        while ((i = (j = t.charAt(x++)).charCodeAt(0))) {
+            var m = i == 46 || (i >= 48 && i <= 57);
+            if (m !== n) {
+                tz[++y] = '';
+                n = m;
+            }
+            tz[y] += j;
+        }
+        return tz;
+    }
+
+    function alphanumForward(a, b) {
+        var aa = chunkify(a.title);
+        var bb = chunkify(b.title);
+
+        for (let x = 0; aa[x] && bb[x]; x++) {
+            if (aa[x] !== bb[x]) {
+                var c = Number(aa[x]),
+                    d = Number(bb[x]);
+                if (c == aa[x] && d == bb[x]) {
+                    return c - d;
+                } else return aa[x] > bb[x] ? 1 : -1;
+            }
+        }
+        return aa.length - bb.length;
+    }
+
+    function alphanumBackward(a, b) {
+        var bb = chunkify(a.title);
+        var aa = chunkify(b.title);
+
+        for (let x = 0; aa[x] && bb[x]; x++) {
+            if (aa[x] !== bb[x]) {
+                var c = Number(aa[x]),
+                    d = Number(bb[x]);
+                if (c == aa[x] && d == bb[x]) {
```

Comment:
error TS2367: This condition will always return 'false' since the types 'number' and 'string' have no overlap.